user/src/com/google/gwt/user/server/rpc/RPCServletUtils.java

```diff
@@ -331,50 +333,23 @@ public static boolean shouldGzipResponseContent(HttpServletRequest request,
    * @throws IOException if reading, writing, or closing the response's output
    *           stream fails
    */
-  public static void writeResponse(ServletContext servletContext,
-      HttpServletResponse response, String responseContent, boolean gzipResponse)
+  public static Writer createWriterForResponse(ServletContext servletContext,
+      HttpServletResponse response, boolean gzipResponse)
```

Comment:
javadoc is now incorrect, responseContent was removed.

That said, this is likely to register as a breaking API change still, so it might make more sense to restore the old version.

Reply:
Are you suggesting you'd like to preserve writeResponse although it is not called anymore, along the lines of what you wrote about splitting the code into two "branches" where the branch for protocol version 9 wouldn't keep any compatibility?

We can keep writeResponse in addition to the new createWriterForResponse, but it would be dead code in the framwork and at least would require deprecation. I'm not sure, either, how any potential use of writeResponse by some framework or specialization would blend into the modified approach to how the writer is created. Any thoughts?

Reply:
2ed50ad28e5a1c4016054d626df121339e0b134d removes the @param line for now.

Reply:
It appears that anything in a `.server.` package is considered "not checked as being part of the API", so the default config won't flag this change... Also it appears I failed to push the actual new config file, so I'll correct that, and maybe discuss widening the "public" api, at least as far as automation is concerned.